### ISSUE TITLE
Implement priors for est="vae"

### ISSUE THREAD

**Issue by mattfidler** [nlmixr2/nlmixr2est#935]
Implement prior distributions for `est="vae"`.

## Why

The VAE trains on an ELBO, so a parameter prior is a natural extra term rather than a bolt-on. The objective is assembled in one line — `vaeDecoderElboStep_()` (`src/inner.cpp:18778`), at **`src/inner.cpp:18847`**:

```cpp
double DKL = pz - qz, loss = pxz + alphaKL * DKL;
```

The training loop is R-side (`R/vaeFit.R:163`, `.vaeElboStep()` at `R/vaeFit.R:61`), and the reported `objective` is deliberately *not* set by the VAE (`R/vaeOutput.R:454-456`) — it comes from an eval-only FOCEi pass in `.vaeToFit()` (`R/vaeOutput.R:361`).

## What a fix looks like

1. Add `log p(theta)` to the loss at `src/inner.cpp:18847`, with the gradient going into `grads`, using the kernel from #929.
2. Decide how the prior interacts with `alphaKL`. The KL term is already scaled by a tempering factor; whether a parameter prior should be scaled with it or held at full weight is a real choice that affects the fit, and it should be stated explicitly rather than falling out of wherever the term happens to be added.
3. Because the reported objective comes from a FOCEi pass, it inherits #931 — check the two are consistent, or the reported number will disagree with what training optimized (the same hazard as #932).

Depends on #929, #930.

**Comment by mattfidler** [comment]
Updating for what actually landed upstream: [nlmixr2/rxode2#1270](https://github.com/nlmixr2/rxode2/pull/1270) (merged) built the shared kernel in rxode2, as three methods -- `"general"` (default), `"nwpri"`, `"tnpri"` -- rather than the `"none"`/`"normal"`/`"nwpri"`/`"all"` gate levels this issue and #929/#930 originally assumed. See the updated #929 for the full mapping and current plan. Still blocked on #929's plumbing (spec carried through `foceiControl`/`op_focei`), which is being picked up now together with #931 (the FOCEI family, the smallest consuming case).
